crates/rolldown_error/src/lib.rs

```diff
@@ -10,6 +10,8 @@ use sugar_path::SugarPath;
 
 pub use crate::{build_error::BuildError, event_kind::EventKind};
 
+pub type Result<T> = std::result::Result<T, BuildError>;
```

Comment:
```suggestion
pub type BuildResult<T> = std::result::Result<T, BuildError>;
```

I prefer don't be conflicted with official `Result` of rust std.

Reply:
I don't think is correct. Like rustc has `std::result::Result`, `std::io::Result`, etc. It is better has `rolldown:result`.

Reply:
I think this not a question of correct or wrong. `std::result::Result` is included in the scope by default. You could use it without import statements. Using `BuildResult` to avoid conflict make you could write official `Result` and `BuildResult` together. Otherwise you need write `rolldown_error::Result<>` explicitly.

As I said above, this is not a not a question of correct or wrong. It's about programing style and I think it's a better style due to above reasons. I'm open to other styles, if they prove to be more beneficial, but I hardly, in this cause, think "rustc did it this way" as a strong argument.